project-koku/koku
# [COST-382] RBAC works on tag value searching/node restrictions

RBAC wasn't working properly on the tags endpoint when it came to access restrictions based on node or when searching for a value. The following change should fix that.

Testing:
Load in some data with `make create-test-customer` and `make load-test-customer-data`

In order to test the changes, the access in your .env file will need to be changed.

1. Testing a restriction based on node:
     - set access to: `"access": {"openshift.node":{"read": ["aws_compute1"]}}`
     - hit the following endpoint: `http://localhost:8000/api/cost-management/v1/tags/openshift/app/?filter[value]=c`
     - you should only see catalog as a result

2. Testing a restriction based on cluster:
     - set access to: `"access": {"openshift.cluster":{"read": ["my-ocp-cluster-2"]}}`
     - hit the following endpoint: `http://localhost:8000/api/cost-management/v1/tags/openshift/app/?filter[value]=b`
     - you should see banking mobile and sombrero as a result

3. Testing a restriction based on a project:
     - set access to: `"access": {"openshift.project":{"read": ["cost-management"]}}`
     - hit the following endpoint: `http://localhost:8000/api/cost-management/v1/tags/openshift/app/?filter[value]=c`
     - you should only see cost as a result

4. Remove the access restriction in your .env and try the filters again and see that there are many more results without the access restriction to verify the RBAC restriction was actually not including things in the results.
